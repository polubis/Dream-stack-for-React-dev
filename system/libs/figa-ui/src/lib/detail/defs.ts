import type { ReactNode } from 'react';

type DetailProps  ={
  className?: string;
  label: ReactNode;
  value: ReactNode;
}

export type { DetailProps };
