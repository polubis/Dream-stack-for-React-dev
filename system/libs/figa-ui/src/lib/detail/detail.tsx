import styled from 'styled-components';
import { Font } from '../font';
import type { DetailProps } from './defs';
import c from 'classnames';
import { tokens } from '../theme-provider';

const Container = styled.div`
  display: grid;
  grid-template-columns: auto;
  gap: ${tokens.spacing[50]};
`;

const Detail = ({ className, label, value }: DetailProps) => {
  return (
    <Container className={c('detail', className)}>
      <Font variant="b2">{label}</Font>
      <Font variant="b1" motive="primary">
        {value}
      </Font>
    </Container>
  );
};

export { Detail };
