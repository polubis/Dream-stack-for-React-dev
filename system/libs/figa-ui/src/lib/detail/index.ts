export * from './detail';
export * from './defs';
