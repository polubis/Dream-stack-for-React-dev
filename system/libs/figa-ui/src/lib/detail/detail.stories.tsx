import type { Story, Meta } from '@storybook/react';

import { Detail } from './detail';
import { Box } from '../box';

export default {
  component: Detail,
  title: 'Detail',
} as Meta;

const Template: Story = () => {
  return (
    <Box padding={[300, 300, 300, 300]}>
      <Detail value="10" label="Number of users" />
    </Box>
  );
};

export const Default = Template.bind({});
Default.args = {};
