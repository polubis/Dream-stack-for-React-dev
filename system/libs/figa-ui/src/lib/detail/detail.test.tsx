import { render } from '@testing-library/react';
import { Detail } from './detail';

describe('Detail can be used when: ', () => {
  it('[FRAGILE] assigns classes', () => {
    const { container, asFragment } = render(
      <Detail
        className="my-class"
        label="My custom label"
        value="My custom value"
      />
    );

    const component = container.querySelector('.detail');

    expect(component?.className).toContain('detail my-class');
    expect(asFragment()).toMatchSnapshot();
  });
});
