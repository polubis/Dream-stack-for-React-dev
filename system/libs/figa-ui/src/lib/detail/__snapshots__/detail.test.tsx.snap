// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail can be used when:  [FRAGILE] assigns classes 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe kEOMNP detail my-class"
  >
    <p
      class="font b2 default"
    >
      My custom label
    </p>
    <p
      class="font b1 primary"
    >
      My custom value
    </p>
  </div>
</DocumentFragment>
`;
